### DISCUSSION TITLE
Prometheus Exporter

### DISCUSSION THREAD

**Discussion by GoutamVerma** [ACINQ/eclair#2280] [Developers]
Hello Everyone,

In order to implement exporter for Prometheus,
-First I have taken a look at implementation of Kamon(Kanela agent) and traced up the coding part of it. 
-Then Instead of creating new object. I have added the Prometheus code stuff in Metrics and Monitoring object.
-At initial point, I will scrape metrics of "balance".

Note: All Prometheus libraries and classes are stored in `io.prometheus.client.Gauge` and in our code base we have another package named "io" which consist client, clientspawner, etc. The problem is when I build our complete code It will show error
```
error: object prometheus is not a member of package io  
[ERROR] import io.prometheus.client.Gauge
```
![Untitled](https://user-images.githubusercontent.com/66783850/169660099-180b6cc3-5752-421c-83f5-cdc6df82ef4a.jpg)

According to me this error is due to naming convection or dual name of name thing.
![image](https://user-images.githubusercontent.com/66783850/169659538-3896164b-51a4-44aa-b9a7-78274c787418.png)

Please comments your views!

**Comment by t-bast** [comment]
> in our code base we have another package named "io" which consist client, clientspawner, etc

That doesn't matter at all, the full namespace is `fr.acinq.eclair.io` so it doesn't clash in any way with `io.prometheus.xxx`.
I suspect that the error you're having is because you didn't add the prometheus package to the `pom.xml` file, did you?
Did you add the `<dependency>` tag to `pom.xml` as indicated in the Kamon documentation here: https://kamon.io/docs/latest/reporters/prometheus/ ? Don't forget that we're using `Maven`, not `sbt`, so you need to follow the `Maven` documentation.

**Reply by GoutamVerma** [reply]
Yes mentor, I have already added the package to the `pom.xml` file and in `project structure`.
But now the issue is resolved and yes I have taken a look at https://kamon.io/docs/latest/reporters/prometheus/ .

And also got an exposed metrics on port 9095. But it is exposed by kamon-prometheus
![image](https://user-images.githubusercontent.com/66783850/169965133-13285faa-9de5-483f-968e-cfad0482dd5b.png)
 Metrics on port: 9095
![image](https://user-images.githubusercontent.com/66783850/169965192-3a667752-bfe6-4ccd-804a-0b79f38b5a78.png)

**Reply by GoutamVerma** [reply]
One more thing mentor for exposing metrics through client_java. There is three options HTTP, Exporting to a Pushgateway and Bridges.
And I am using HTTP for our project
